src/shared/utils/utils.ts

```diff
@@ -92,6 +92,31 @@ export const getMerkleRoot = (document: any): string => {
   }
 };
 
+export const getTargetHash = (document: any): string => {
+  switch (true) {
+    case isWrappedV2Document(document):
+      return document.signature.targetHash;
+    case isWrappedV3Document(document):
+      return document.proof.targetHash;
+    default:
+      throw new Error(
+        "Unsupported document type: Only can retrieve target hash from wrapped OpenAttestation v2 & v3 documents."
+      );
+  }
+};
+
+export const getAssetId = (document: any): string => {
```

Comment:
if you extract the code for checking if it is a transferable asset you can simplify this function down to 

if (isTransferableAsset(document) { return getTargetHash(document) }
else { throw error }

this way you don't need to maintain two identical code in getTargetHash and here for retrieving the targetHash

Reply:
other than that lgtm

Reply:
ok refactored accordingly.